operators/src/source/gdal_source/mod.rs

```diff
@@ -733,7 +734,27 @@ where
             loading_info.info
         };
 
-        let source_stream = stream::iter(loading_iter);
+        let loading_inter_checked = loading_iter.with_position().map(move |(pos, ele)| {
+            let q_time = query.time_interval;
+            match (ele, &pos) {
+                (Ok(e), Position::First) if q_time.start() < e.time.start() => {
+                    warn!("The data provider does not cover the query start. Query time {}, First element time {}", q_time, e.time);
+                    Err(error::GdalSourceError::MissingTemporalDataAtQueryStart{query_time: q_time, first_element_time: e.time}.into())
+                },
+                (Ok(e), Position::Last) if q_time.end() > e.time.end() => {
+                    warn!("The data provider does not cover the query end. Query time {}, Last element time {}", q_time, e.time);
+                    Err(error::GdalSourceError::MissingTemporalDataAtQueryEnd{query_time: q_time, last_element_time: e.time}.into())
+                },
+                (Ok(e), Position::Only) if q_time.start() < e.time.start() || q_time.end() > e.time.end() => {
+                    warn!("The data provider does not cover the query end. Query time {}, Only element time {}", q_time, e.time);
```

Comment:
why do you only check the end if there is only one element?

Reply:
i removed that and use the filler instead